### PR TITLE
fix(lsp): support mixed encoding for lsp clients

### PR DESCRIPTION
**Pull request by tris203**
## Description

Fixes a warning notification for the first hover/signature help notification on nightly now that `offset_encoding` is required for `make_position_params`

### CODE REVIEW

File: lua/noice/lsp/init.lua [hunk 1]
```diff
@@ -51,15 +51,15 @@ function M.scroll(delta)
 end
 
 function M.hover()
-  ---@diagnostic disable-next-line: missing-parameter
-  local params = vim.lsp.util.make_position_params()
-  vim.lsp.buf_request(0, "textDocument/hover", params, require("noice.lsp.hover").on_hover)
+  vim.lsp.buf_request(0, vim.lsp.protocol.Methods.textDocument_hover, function(client)
```
Comment (folke) [inline lua/noice/lsp/init.lua:54]: `vim.lsp.protocol.Methods` is not available on Neovim < 0.10.

**Review by folke — CHANGES_REQUESTED** [review]
Apart from the availability of `vim.lsp.protocol`, I also don't think passing a function as params works on Neovim <= 0.9?

**Comment by tris203** [comment]
I didn't read the README properly

I had always assumed noice was nightly only, not nightly recommended 

I can add a shim for back compat this evening

**Comment by tris203** [comment]
Add a shim, and removed the reference to `vim.lsp.protocol.Methods`

**Review by folke — APPROVED** [review]

**Pull request merged**